src/daft-micropartition/src/micropartition.rs

```diff
@@ -359,13 +359,21 @@ fn materialize_scan_task(
 impl MicroPartition {
     /// Create a new "unloaded" MicroPartition using an associated [`ScanTask`]
     ///
-    /// Schema invariants:
+    /// Invariants:
     /// 1. Each Loaded column statistic in `statistics` must be castable to the corresponding column in the MicroPartition's schema
+    /// 2. Creating a new MicroPartition with a ScanTask that has any filter predicates or limits is not allowed and will panic
     pub fn new_unloaded(
         scan_task: Arc<ScanTask>,
         metadata: TableMetadata,
```

Comment:
Is this necessary as an argument? Shouldn't we be able to get this from the ScanTask?